snapshots/output/syntax/src/object-literals-arrow-function.ts

```diff
@@ -0,0 +1,88 @@
+// < definition syntax 1.0.0 src/`object-literals-arrow-function.ts`/
+
+import { Option } from './reusable-types'
+//       ^^^^^^ reference syntax 1.0.0 src/`reusable-types.ts`/Option#
+//                     ^^^^^^^^^^^^^^^^^^ reference syntax 1.0.0 src/`reusable-types.ts`/
+
+interface Foobar {
+//        ^^^^^^ definition syntax 1.0.0 src/`object-literals-arrow-function.ts`/Foobar#
+  foobar: number
+//^^^^^^ definition syntax 1.0.0 src/`object-literals-arrow-function.ts`/Foobar#foobar.
+}
+
+export function hasArrowFunctionParameter(
+//              ^^^^^^^^^^^^^^^^^^^^^^^^^ definition syntax 1.0.0 src/`object-literals-arrow-function.ts`/hasArrowFunctionParameter().
+  something: number,
+//^^^^^^^^^ definition syntax 1.0.0 src/`object-literals-arrow-function.ts`/hasArrowFunctionParameter().(something)
+  fn: (foobar: Foobar) => Foobar
+//^^ definition syntax 1.0.0 src/`object-literals-arrow-function.ts`/hasArrowFunctionParameter().(fn)
+//     ^^^^^^ definition local 1
+//             ^^^^^^ reference syntax 1.0.0 src/`object-literals-arrow-function.ts`/Foobar#
+//                        ^^^^^^ reference syntax 1.0.0 src/`object-literals-arrow-function.ts`/Foobar#
+): Foobar {
+// ^^^^^^ reference syntax 1.0.0 src/`object-literals-arrow-function.ts`/Foobar#
+  return fn({ foobar: 42 + something })
+//       ^^ reference syntax 1.0.0 src/`object-literals-arrow-function.ts`/hasArrowFunctionParameter().(fn)
+//            ^^^^^^ reference syntax 1.0.0 src/`object-literals-arrow-function.ts`/Foobar#foobar.
```

Comment:
Woo hoo, this is such a significant leap for navigation quality. Closest  huge gap against tsserver